### PR TITLE
Add tests of geo applications

### PR DESCRIPTION
**Pull request by ggardet**
- Verification run: 
  * console test: https://openqa.opensuse.org/t1095286
  * desktop test: https://openqa.opensuse.org/t1090760
- Needles:
  * https://github.com/os-autoinst/os-autoinst-needles-opensuse/commit/3f221c5cf0391037e6b61395124e59f1f2db043e
  * https://github.com/os-autoinst/os-autoinst-needles-opensuse/commit/a57aca734e94193b8bb41295d30e8fcc82054209

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -1585,6 +1585,14 @@ sub load_extra_tests_qemu {
     loadtest "qemu/user" if is_opensuse;
 }
 
+sub load_extra_tests_geo_console {
+    loadtest "appgeo/gdal" if is_tumbleweed;
+}
+
+sub load_extra_tests_geo_desktop {
+    loadtest "appgeo/qgis" if is_tumbleweed;
```
Comment (foursixnine) [inline lib/main_common.pm]: YAY! :D

File: tests/appgeo/qgis.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    if (check_screen('qgis-welcome')) {
+        # Close tip of day
+        wait_screen_change { send_key 'esc'; };
+    }
+
+    # Check we have the qgis main window
+    assert_screen('qgis');
+
+    # Close QGIS
+    send_key 'alt-f4';
```
Comment (foursixnine) [inline tests/appgeo/qgis.pm]: Maybe extend it by opening a file?

File: tests/appgeo/gdal.pm [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+use strict;
+use warnings;
+use base "consoletest";
+use testapi;
+use utils;
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
```
Comment (foursixnine) [inline tests/appgeo/gdal.pm]: I'd use serial console here (it's just faster)

File: tests/appgeo/qgis.pm (hunk 2 above)
Comment (ggardet) [reply tests/appgeo/qgis.pm]: Maybe later. Currently I am adding simple tests which will be improved later. ;)

File: tests/appgeo/gdal.pm (hunk 3 above)
Comment (ggardet) [reply tests/appgeo/gdal.pm]: Not sure how to do this. And, I need root permission for zypper.

Comment (asmorodskyi) [reply tests/appgeo/gdal.pm]: serial console is runs under root , maybe you can try it out ?

Comment (ggardet) [reply tests/appgeo/gdal.pm]: > serial console is runs under root , maybe you can try it out ?

Do you have an example how to do it, please?

Comment (asmorodskyi) [inline tests/appgeo/gdal.pm]:
```suggestion
 $self->select_serial_terminal;
```
also you will need to define env. variable VIRTIO_CONSOLE=1

Comment (asmorodskyi) [reply tests/appgeo/gdal.pm]: and of course you need 
`use serial_terminal;`

Comment (ggardet) [reply tests/appgeo/gdal.pm]: > also you will need to define env. variable VIRTIO_CONSOLE=1

Is it available for all workers?

Comment (asmorodskyi) [reply tests/appgeo/gdal.pm]: yes

Comment (asmorodskyi) [reply tests/appgeo/gdal.pm]: let's keep it clear - at least from my side I don't see this as must to have . It is just **nice** to have. So if with whatever reason you like to stick to 'root-console' I am fine with that

Comment (foursixnine) [reply tests/appgeo/gdal.pm]: It does dun under root.

Comment (ggardet) [reply tests/appgeo/gdal.pm]: Done

**Comment by ggardet** [comment]
It should be ready to merge now. :)

**Comment by asmorodskyi** [comment]
if you say so :)

**Pull request merged**